server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala

```diff
@@ -1059,7 +1059,7 @@ trait MailboxSetMethodContract {
          |      "notCreated": {
          |        "C42": {
          |          "type": "invalidArguments",
-         |          "description": "'/parentId' property in mailbox object is not valid: Predicate isEmpty() did not fail."
+         |          "description": "'/parentId' property in mailbox object is not valid: Left predicate of ((!(0 < 1) && !(0 > 255)) && \\"\\".matches(\\"^[#a-zA-Z0-9-_]*$$\\")) failed: Predicate taking size() = 0 failed: Left predicate of (!(0 < 1) && !(0 > 255)) failed: Predicate (0 < 1) did not fail."
```

Comment:
this looks like an over-precise check: it expects an error specific to Refined but should not

Reply:
This is over-precise because this error is generated at the JSON deserialization layer, and we likely have no to little context there.

Instead of relying on JSON refined implicit, then we would have to provide separate types deserialization wrapping our knowledge, I imagine.

Reply:
It's not the message that I dislike, it's the assertion, sorry if my comment was confusing. Could you just ignore part of this message?

Reply:
Yes.

(I don't like the assertion neither)

But you are right that the description is cryptic ad lacks context. Hopefully it is now prefixed with a human readable message.